config-overrides/index.js

```diff
@@ -0,0 +1,159 @@
+    else delete config.devtool
+    config.optimization.minimize = false
+    config.entry = {
+        'options-page': src('./src/index.tsx'),
+        'content-script': src('./src/content-script.ts'),
+        'background-service': src('./src/background-service.ts'),
+        'injected-script': src('./src/extension/injected-script/index.ts'),
+        popup: src('./src/extension/popup-page/index.tsx'),
+        qrcode: src('./src/web-workers/QRCode.ts'),
+    }
+    if (env !== 'development') delete config.entry.devtools
+
+    // Let bundle compatible with web worker
+    config.output.globalObject = 'globalThis'
+    config.output.filename = 'js/[name].js'
+    config.output.chunkFilename = 'js/[name].chunk.js'
+
+    // We cannot do runtimeChunk because extension CSP disallows inline <script>
+    config.optimization.runtimeChunk = false
+    config.optimization.splitChunks = false
+
+    // Dismiss warning for gun.js
+    config.module.wrappedContextCritical = false
+    config.module.exprContextCritical = false
+    config.module.unknownContextCritical = false
+
+    config.plugins = config.plugins.filter(x => x.constructor.name !== 'HtmlWebpackPlugin')
+    /**
+     * @param {HtmlWebpackPlugin.Options} options
+     */
+    function newPage(options = {}) {
+        return new HtmlWebpackPlugin({
+            templateContent: fs.readFileSync(path.join(__dirname, './template.html'), 'utf8'),
+            inject: 'body',
+            ...options,
+        })
+    }
+
+    if (process.argv.indexOf('--firefox') !== -1) {
+        config.plugins.push(new WebExtPlugin({ sourceDir: dist }))
+    }
+
+    if (process.argv.indexOf('--firefox-android') !== -1) {
+        config.plugins.push(new WebExtPlugin({ sourceDir: dist, target: 'firefox-android' }))
+    }
+
+    config.plugins.push(
+        newPage({ chunks: ['options-page'], filename: 'index.html' }),
+        newPage({ chunks: ['background-service'], filename: 'background.html' }),
+        newPage({ chunks: ['popup'], filename: 'popup.html' }),
+        newPage({ chunks: ['content-script'], filename: 'generated__content__script.html' }),
+    )
+
+    config.plugins.push(
+        new webpack.BannerPlugin(
+            'Maskbook is a open source project under GNU AGPL 3.0 licence.\n\n\n' +
+            'More info about our project at https://github.com/DimensionDev/Maskbook\n\n' +
+            'Maskbook is built on CircleCI, in which all the building process is available to the public.\n\n' +
+            'We directly take the output to submit to the Web Store. We will integrate the automatic submission\n' +
+            'into the CircleCI in the near future.',
+        ),
+    )
+
+    // Write files to /public
+    if (env === 'development') {
+        config.plugins.push(
+            new (require('copy-webpack-plugin'))(
+                [...polyfills.map(from => ({ from, to: publicPolyfill })), { from: publicDir, to: dist }],
```

Comment:
Sure. It said that polyfill is not defined, seems like you missed a import statement here.

Reply:
It should have been removed.